generator/test/good_test.dart

```diff
@@ -0,0 +1,264 @@
+            '      SchemaProperty(\'age\', RealmPropertyType.int),\n'
+            '    ]);\n'
+            '  }\n'
+            '}\n'
+            '',
+      },
+      reader: await PackageAssetReader.currentIsolate(),
+    );
+  });
+
+  test('required argument', () async {
+    await testBuilder(
+      generateRealmObjects(),
+      {
+        'pkg|lib/src/test.dart': r'''
+import 'package:realm_common/realm_common.dart';
+
+part 'test.g.dart';
+
+@RealmModel()
+class _Person {
+  @PrimaryKey()
+  late final String name;
+}''',
+      },
+      outputs: {
+        'pkg|lib/src/test.realm_objects.g.part': '// **************************************************************************\n'
+            '// RealmObjectGenerator\n'
+            '// **************************************************************************\n'
+            '\n'
+            'class Person extends _Person with RealmObject {\n'
+            '  static var _defaultsSet = false;\n'
+            '\n'
+            '  Person(\n'
+            '    String name,\n'
+            '  ) {\n'
+            '    _defaultsSet = _defaultsSet || RealmObject.setDefaults<Person>({});\n'
+            '    RealmObject.set(this, \'name\', name);\n'
+            '  }\n'
+            '\n'
+            '  Person._();\n'
+            '\n'
+            '  @override\n'
+            '  String get name => RealmObject.get<String>(this, \'name\') as String;\n'
+            '\n'
+            '  static SchemaObject get schema => _schema ??= _initSchema();\n'
+            '  static SchemaObject? _schema;\n'
+            '  static SchemaObject _initSchema() {\n'
+            '    RealmObject.registerFactory(Person._);\n'
+            '    return const SchemaObject(Person, [\n'
+            '      SchemaProperty(\'name\', RealmPropertyType.string, primaryKey: true),\n'
+            '    ]);\n'
+            '  }\n'
+            '}\n'
+            ''
+      },
+      reader: await PackageAssetReader.currentIsolate(),
+    );
+  });
+
+  test('list initialization', () async {
+    await testBuilder(
+      generateRealmObjects(),
+      {
+        'pkg|lib/src/test.dart': r'''
+import 'package:realm_common/realm_common.dart';
+
+part 'test.g.dart';
+
+@RealmModel()
+class _Person {
+  late final List<_Person> children;
+}''',
+      },
+      outputs: {
+        'pkg|lib/src/test.realm_objects.g.part': '// **************************************************************************\n'
+            '// RealmObjectGenerator\n'
+            '// **************************************************************************\n'
+            '\n'
+            'class Person extends _Person with RealmObject {\n'
+            '  static var _defaultsSet = false;\n'
+            '\n'
+            '  Person({\n'
+            '    Iterable<Person> children = const [],\n'
+            '  }) {\n'
+            '    _defaultsSet = _defaultsSet || RealmObject.setDefaults<Person>({});\n'
```

Comment:
I suggest we change this a bit. right now its always an assignment. even if _defaultsSet is true already.  
I will create a suggestion for the suggested change

Reply:
No, operator || only evaluates left side, if it is true.

Reply:
But in this case (since it is okay not to call setDefaults, if set is empty), we can get rid of it completely

Reply:
No longer set empty default set, but still keep flag as bool

Reply:
https://dartpad.dev/?id=fac95cee1ad60ee24a2eacfdacb896eb demonstrates behavior of operator||

Reply:
I mean `_defaultsSet = `  is doing an assignment every time an instance is created

Reply:
Okay, sorry. I'll check first.

Reply:
Fixed

Reply:
checking if (_defaultsSet) is good as well, imo.